onnxruntime/core/graph/contrib_ops/contrib_defs.cc

```diff
@@ -3239,6 +3239,41 @@ Input zero_points is stored as uint8_t. If bits <= 4, two zero points are stored
         MatmulWithQuantWeightShapeInference(ctx, in_features, out_features);
       });
 
+  static const char* MatMulBnb4_ver1_doc = R"DOC(
+MatMulBnb4 is a MatMul with weight quantized with 4 bits using either FP4 or NF4 data type (https://arxiv.org/pdf/2305.14314.pdf). It does Matrix Multiplication like MatMul (https://github.com/onnx/onnx/blob/main/docs/Operators.md#matmul) with differences:
+  1. Input B is a 2D constant Matrix. Its input feature count and output feature count are specified by attribute 'K' and 'N'.
+  2. Input B is quantized with 4 bits with quantization data type specified by attribute 'quant_type'. It is transposed, flattened and quantized blockwisely with block size specified by attribute 'block_size'.
+     And block_size is not an arbitrary number and must be a power of 2 and not smaller than 16, like 16, 32, 64, 128,..
+  3. Input B's quantization constants or scales are specified by input 'absmax'.
+
+Input B is stored as uint8_t with shape: [(N * K + 1) / 2].
+Input absmax is stored in same type as original type of B(float32, float16) with shape like: [(N * K + block_size - 1) / block_size].
```

Comment:
>[(N * K + block_size - 1) / block_size]. [](http://example.com/codeflow?start=93&length=40)

just want to double check: is it (K + block_size -1 ) / block_size * N?

Reply:
Since we are flattening out the weight matrix and quantizing the blocks (without caring about the column/row boundaries), it turns out to be `(N * K + block_size - 1) / block_size`.